core/ledger/kvledger/snapshot.go

```diff
@@ -269,8 +275,15 @@ func (p *Provider) CreateFromSnapshot(snapshotDir string) (ledger.PeerLedger, er
 				errors.WithMessage(err, "error while importing data into config history Mgr"),
 			)
 	}
+	btlPolicy := pvtdatapolicy.ConstructBTLPolicy(
+		&mostRecentCollectionConfigFetcher{
+			DeployedChaincodeInfoProvider: p.initializer.DeployedChaincodeInfoProvider,
```

Comment:
Not related to this PR. We need to somehow get rid of this dependency. Had to spend some time to understand the difference between this `ConstructBTLPolicy()` and the following one https://github.com/hyperledger/fabric/blob/bbea1f582f360f2557314dd2aa3abb5c8e33a4b8/core/ledger/kvledger/kv_ledger.go#L101

Reply:
Which dependency are you referring to? `ConstructBTLPolicy` or `DeployedChaincodeInfoProvider`?

Reply:
`DeployedChaincodeInfoProvider`

Reply:
Not sure what you meant by getting rid of this dependency. In general, it's not desired to get rid of the dependency - `DeployedChaincodeInfoProvider`, as the primary purpose of this is to contain the chaincode datamodel details in the chaincode package. However, what we should be doing is that the logic of implicit collection should be separated out from this in a separate dependency. I had started created a separate package for this - https://github.com/hyperledger/fabric/pull/1692